### PR TITLE
chore(upgrade): update env files

### PR DESCRIPTION
**Pull request by xzf0587**

### CODE REVIEW

File: packages/fx-core/templates/core/v3Migration/upgradeReport.md [hunk 1]
```diff
@@ -22,11 +22,12 @@ You can visit this [wiki](https://aka.ms/teams-toolkit-5.0-upgrade#feature-chang
 
 1. Created `teamsapp.yml` and `teamsapp.local.yml` in your project root folder.
 2. Moved environment files in `.fx` to `.env.{env}` in `env` folder.
-3. Moved `templates/appPackage` to `appPackage`, renamed `manifest.template.json` to `manifest.json` and placeholders in it will be updated per the latest Teams Toolkit requirement.
-4. If your project contains file `templates/appPackage/aad.template.json`, it will be moved and renamed as `aad.manifest.json` and placeholders in it will be updated per the latest Teams Toolkit requirement.
-5. Updated `.vscode/tasks.json` and `.vscode/launch.json`.
-6. Updated `.gitignore` to ignore new environment files.
-7. Removed `.fx` folder.
+3. If your project contains file `.fx/states/{env}.userdata`, it will be moved to `.env.{env}.user` in `env` folder.
```
Comment (blackchoey) [inline packages/fx-core/templates/core/v3Migration/upgradeReport.md:25]: the content will be moved to `.env.{env}.user` in `env` folder

Please also request Zhidi to help review the wording

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 2]
```diff
@@ -61,6 +61,8 @@ import {
   getDownloadLinkByVersionAndPlatform,
   getVersionState,
   getTrackingIdFromPath,
+  buildEnvUserFileName,
+  tryExtractEnvForUserdata as tryExtractEnvFromUserdata,
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:65]: can you share why we need to rename it?

Comment (xzf0587) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts]: Fixed it. Thanks.

File: packages/fx-core/templates/core/v3Migration/upgradeReport.md (hunk 1 above)
Comment (xzf0587) [reply packages/fx-core/templates/core/v3Migration/upgradeReport.md]: Fixed it. Thanks.

**Review by blackchoey — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**